### PR TITLE
Feat 2277 - Clean how models/interfaces are declared in front

### LINKED ISSUE #2277: Clean how models/interfaces are declared in front
We should have something like :

.firestore
```
export interface MovieBase<D extends Timestamp | Date> {...}
export type MovieDocument = MovieBase<Timestamp>
```
.model
```
import { MovieBase } from './***.firestore'
export interface Movie extends MovieBase<Date> {
  specificFrontEndStuffIfNeeded;
}
```

To avoid switching between 3 files to find  "MovieDocumentWithDate"

### PR DESCRIPTION
**Pull request by phpgeekfr**
fix #2277

### CODE REVIEW

File: libs/notification/src/lib/+state/notification.firestore.ts [hunk 1]
```diff
@@ -39,11 +40,11 @@ export interface NotificationOptions {
   organization?: PublicOrganization;
   /** @dev Type of the notification */
   type: NotificationType;
-}
-
-/** Generic informations for a Notification. */
-export interface NotificationDocument<D extends Date | firestore.Timestamp = firestore.Timestamp> extends NotificationOptions {
```
Comment (phpgeekfr) [inline libs/notification/src/lib/+state/notification.firestore.ts:45]: @GrandSchtroumpf I don't understand why `NotificationDocument` was previously declared with `<D extends Date | firestore.Timestamp = firestore.Timestamp>` instead using firestore.Timestamp only .  Do you have a hint ?
This is only used in backend so it should always handle only timestamps

**Comment by phpgeekfr** [comment]
I did not touch `distributionRights` since it is going to be reworked from scratch

File: libs/notification/src/lib/+state/notification.firestore.ts (hunk 1 above)
Comment (GrandSchtroumpf) [reply libs/notification/src/lib/+state/notification.firestore.ts:45]: Notifications are only used in the backend ? In the frontend we don't do something like that ?
```typescript
interface Notification extends NotificationDocument<Date> {}
```
If not then you can force it to Timestamp

Comment (phpgeekfr) [reply libs/notification/src/lib/+state/notification.firestore.ts:45]: NotificationDocument are only used in backend and  for front we have `export interface Notification extends NotificationBase<Date> {`

**Review by fritzschoff — APPROVED** [review]
lgtm

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**